ldcoolp/curation/retrieve.py

```diff
@@ -34,32 +35,19 @@ def private_file_retrieve(url, filename=None, token=None, url_open=False,
     if isinstance(log, type(None)):
         log = log_stdout()
 
-    if not url_open:
-        if token:
-            opener = build_opener()
-            opener.addheaders = [('Authorization', f'token {token}')]
-            install_opener(opener)
-        else:
-            opener = build_opener()
-            install_opener(opener)
-
-        try:
-            urlretrieve(url, filename)
-        except HTTPError as error:
-            log.warning(f"Caught an HTTPError: {error}")
-            raise
-    else:
-        req = Request(url)
-        if token:
-            req.add_header('Authorization', f'token {token}')
+    headers = dict()
+    if token:
+        headers['Authorization'] = f'token {token}'
 
-        response = urlopen(req)
-        content = response.read()
-        print(url)
+    try:
+        h = requests.head(url, headers=headers)
+        h.raise_for_status()
 
-        f = open(filename, 'wb')
-        f.write(content)
-        f.close()
+        with requests.get(url, stream=True, headers=headers) as r:
```

Comment:
Where is the chunked reads/writes implementation? I spent a few minutes looking for it because of the the name of the PR. Minor thing but I suggest renaming the PR

Reply:
This is in the above lines of code. The `stream` option will chunk read the file from `requests`. The `shutil.copyfileobj` has built-in chunk writing. See [docs](https://docs.python.org/3/library/shutil.html#shutil.copyfileobj). The default `length` is 16KB.